src/runtime-rs/crates/hypervisor/README.md

```diff
@@ -89,6 +90,214 @@ pub trait Hypervisor: Send + Sync {
+
+2. What DBS does?
+* DBS first run its VMM server.
+    * It creates the jailer path of the VM, setup the channel for thread communication, and start a thread to run its event_loop, since DBS runs as the thread within the process. Other VMM may start another process.
+* DBS then starts its kernel.
+    * It does normal things as a VMM, prepare kernel_params, initrd, image, rootfs, etc. and runs with its kernel started.
+
+---
+
+```rust
+async fn stop_vm(&self) -> Result<()>
+```
+1. What is expected?
+* In general, `stop_vm()` wants you to terminate the VMM from its execution.
+
+2. What DBS does?
+* It sends an stopVM request to DBS VMM server, the VMM server will receive the request and exit its execution from the thread called `vmm_thread`.
+
+---
+
+```rust
+    async fn pause_vm(&self) -> Result<()>;
+```
+1. What is expected?
+* In general, `pause_vm()` wants you to pause the VMM from its execution like it said. The ways to actually pause the VMM differs from VMMs.
+
+2. What DBS does?
+* Not implemented in DBS.
+
+---
+
+```rust
+    async fn resume_vm(&self) -> Result<()>;
+```
+1. What is expected?
+* This is a paired function with `pause_vm()`, it resumes the VMM's execution from the paused state.
+
+2. What DBS does?
+* Not implemented in DBS.
+
+---
+
+```rust
+    async fn save_vm(&self) -> Result<()>;
+```
+1. What is expected?
+* TBD
+
+2. What DBS does?
+* Not implemented in DBS.
+
+---
+    
+```rust
+async fn add_device(&self, device: device::Device) -> Result<()>;
+```
+1. What is expected?
+* This functions is a generic interface of adding devices which implement trait `device::Device`. If the VMM
+is started, the device should be hotplugged. If the VMM have not started, the device should be in some pending list (s.t. it is never forgotten) and plugged when VM starts.
+
+2. What DBS does?
+* If DBS has booted, it hotplugs the device if supported. Else, it puts the devices in a pending list and plugs them when DBS boots.
+* Currently, devices include `Network device`, `VFIO device`, `Block device`, `Vsock`, `ShareFsDevice`, `ShareFsMount`.
+    * DBS treats `ShareFsDevice and ShareFsMount` as devices, but this should be VMM-specific.
+
+---
+
+```rust
+    async fn remove_device(&self, device: device::Device) -> Result<()>;
+```
+1. What is expected?
+* In general, `remove_device()` asks the VMM to remove the device which implemented trait `device::Device`.
+
+2. What DBS does?
+* DBS currently only supports remove block drive, other devices are not supported to be removed.
+
+---
+    
+```rust
+    async fn get_agent_socket(&self) -> Result<String>;
+```
+1. What is expected?
+* In Kata Containers, runtime talks to agent through a VSock. In the agent, the agent use VMADDR_CID_ANY to bind to the vsock.
+> Note: Therefore, the VSock has to be inserted before the agent start. Note that the agent is booted by systemd in guest kernel.
```

Comment:
not always booted by systemd.
You could take a look at https://github.com/kata-containers/kata-containers/tree/main/tools/osbuilder/rootfs-builder for alpine rootfs information

Reply:
This is a really nice catch! Thanks.